docs/branching.md

```diff
@@ -0,0 +1,98 @@
+# Branching and deployments model
+
+This document aims to discuss and document the relations between branches and deployments of `elixir-omg`, with respect to branches and deployments of `plasma-contracts`.
+
+It is a refinement of the [OIP4 branching model](https://github.com/omisego/OIP/blob/master/0004-ewallet-release-and-versioning.md), applicable to `elixir-omg` and `plasma-contracts` versioning.
+
+Rationale:
+- the history and the relations between the versions are readable and simple to understand
+- we can predictably sync our respective watchers/run child chain servers against deployed contracts
+- we can move with the `master` branch quickly
+
+
+## Dependency Rules for elixir-omg
+
+### For mix.exs
+- `elixir-omg/master` will point to the `plasma-contracts/master`
+- A release branch in `elixir-omg` will point to the corresponding release branch in `plasma-contracts`. For example:
+  - `elixir-omg/v0.1` -> `plasma-contracts/v0.1`
+
+### For mix.lock
+- Always points to a specific SHA that in the history of the `plasma-contracts` branch referenced in `mix.exs`
+
+
+## Deployment Scenarios
+
+### 1 - Single production deployment, ongoing development
+
+This is the active scenario most of the time.
+
+Branches and environments:
+- `master` is automatically deployed to **development** environment
+- `v0.1` is automatically deployed to **staging-v0-1** environment
```

Comment:
nitpick: Is this a set environment naming convention? how do we feel about `staging/v0.1` as the environment name? Is that valid?

Reply:
That sounds reasonable to me

Reply:
Not tied to this environment name format but was intending to be able to use them as subdomains